### DISCUSSION TITLE
CDK Deployment

### DISCUSSION THREAD

**Discussion by romeosunio** [aws/aws-cdk#23366] [Q&A]
Getting this error when deploying cdk via githubflow
```Error [ValidationError]: 1 validation error detected: Value at 'tags.1.member.value' failed to satisfy constraint: Member must have length greater than or equal to 1```

But never encountered this issue when doing a `cdk deploy` from my terminal. 

Thoughts?

**Comment by peterwoodworth** [comment]
Sorry no one got to this, could you provide a bit more context if this is still a issue? I'm not sure in what context `tags.1.member.value` is coming into play

**Reply by kauer2** [reply]
Have just run into the same issue. Using this to tag:

`cdk.Tags.of(scope).add(tag.key, tag.value, tag.props);`

If `tag.value` value is an empty string, CDK 2.159.1 throws this error:

`validation error detected: Value '' at 'tags.13.member.value' failed to satisfy constraint: Member must have length greater than or equal to 1`

This validation check would seem to be misguided, as AWS generally seems perfectly fine with empty tag values. We have older resources, tagged at the time using CDK and using the same code, which do have empty string values. Maybe older constructs? I went back as far as 2.133.0 and saw the same thing.

Here is a minimal program that demonstrates the issue. Build with `cdk deploy --profile=your_profile`:

```
import 'source-map-support/register';
import * as cdk from 'aws-cdk-lib';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import { Construct } from 'constructs';

const app = new cdk.App();

class TagtestStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new ec2.SecurityGroup(this, "sg", {
      vpc: ec2.Vpc.fromLookup(this, "vpc", {
        isDefault: true
      })
    }) ;
  }
}

const t = new TagtestStack(app, 'TagtestStack', {
  env: {
    account: process.env.CDK_DEFAULT_ACCOUNT,
    region: process.env.CDK_DEFAULT_REGION
  },
});

cdk.Tags.of(t).add('AKey', '');
```
